docs/release.md

```diff
@@ -27,85 +37,258 @@ release, those are:
-<kubevirt-dev@googlegroups.com>.
+When you create this token the only permission you need to give it is the
+ability to access github repositories. That's it.
 
-The `hack/release-announce.sh` script should be used to generate the
-announce email.
-Using this script ensures consistency and documentation of the release
-process and content between releases.
+[Instructions for creating access token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token)
+
+**Place your token in a text file such as** ```${HOME}/github-api-token```
+
+## Release Tool Usage Examples
+
+Once you have your credentials in files, the kubevirt release tool handles
+all the rest. All you need to do is provide your credentials and tell the tool
+what release you want to make.
+
+Place the paths to your credential files in the following environment variables.
+
+```
+$ export GPG_PRIVATE_KEY_FILE="${HOME}/gpg-private"
+$ export GPG_PASSPHRASE_FILE="${HOME}/gpg-passphrase"
+$ export GITHUB_API_TOKEN_FILE="${HOME}/github-api-token"
+```
 
+Now you can use the release tool to do whatever you'd like. Note that you can
+use the ```--dry-run=true``` argument to test a change before executing it.
 
-Topic: Signing
---------------
-The workflows below use signed tags. The following link describes what you
-need to do in order to setup correct signing for tags (and commits) on your
-local machine and in GitHub, in order to get the commits verified.
+#### Example: creating a new release branch with the initial release candidate v0.31.0-rc.1
+```
+$ hack/release.sh --new-branch release-0.31 --new-release v0.31.0-rc.1 --dry-run=false
+```
 
-https://help.github.com/articles/adding-a-new-gpg-key-to-your-github-account/
+#### Example: Creating a new rc v0.31.0-rc.1
+```
+$ hack/release.sh --new-release v0.31.0-rc.1 --dry-run=false
+```
 
-Setting up and using signing is mandatory for any release.
+#### Example: Promoting a release candidate v0.31.0-rc-1 to official v0.30.0 release
+```
+$ hack/release.sh --promote-release-candidate v0.31.0-rc-1 --dry-run=false
+```
 
+#### Example: Creating a patch release v0.31.1. The branch will automatically be detected
+```
+$ hack/release.sh --new-release v0.31.1 --dry-run=false
+```
 
-Releasing minor versions
-------------------------
+## Creating New Minor Releases
 The release process is mostly automatic and consists of the following steps:
 
-1. Run the `hack/release-announce.sh` script which runs functional tests,
-   generates some stats and outputs the announce text.
-2. Tag the commit using `git evtag sign $TAG` (which is a signed and annotated
-   tag)
-   1. Provide the announce text as part of the tag commit message
-3. Push the tag to github `git push git@github.com:kubevirt/kubevirt.git $TAG`
+1. Create the branch and initial RC.
+
+   ```
+   $ hack/release.sh --new-branch $RELEASE_BRANCH --new-tag ${TAG}.rc.1
+   ```
+
+2. Wait 5 business days
```

Comment:
there is any reason behind waiting 5 business days? why not 10 or less?

maybe it is better to put something like
> The promotion if rc.1 will occur if we don't find any blockers within 5 business days.

but what happens with an rc.2 or rc.x needs to be cut?

thanks and sorry for the intrusion :)

Reply:
> there is any reason behind waiting 5 business days? why not 10 or less?

The 5 days is a general guideline we're working with as a starting point. If we messed something critical up (rare), it is usually identified in less that 5 business days due to flaky functional tests.

This delay is flexible though. Maybe in the future we shorten it, or lengthen it based on how all this works in practice. 

> but what happens with an rc.2 or rc.x needs to be cut?

5 business days after every rc.  so if rc.2 is cut, then the general policy is to wait another 5 business days until promoting.

Example.
- Monday 1st rc.1 is cut,
- Wednesday 3rd rc.2 is cut
- five business days later, rc.2 is promoted

Reply:
thanks!